### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

Structure and type for the database

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [ ] Frontend
- [X] Backend
- [ ] Other: .....

## General checklist:

- [X] My code follows the style guidelines of this project
- [X] I ran `bun run format`/`bun run lint` before commit
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation where needed
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] I have updated `develop` and merged to my branch before submitting pull request
- [X] My pull request generate no conflicts with `develop` branch
- [X] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [X] I followed guidelines for naming `Typescript` variables
- [X] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by ElBarto444 — APPROVED** [review]
C'est good

File: packages/types/src/database.ts [hunk 1]
```diff
@@ -1,14 +1,122 @@
...
+
+export interface PictureTable extends Omit<ProfileTable, 'name'> {}
+
+export interface HobbyTable extends Omit<ProfileTable, 'logo_path'> {
+  category_id: number;
+}
+
+export interface CityTable extends Omit<ProfileTable, 'name'> {
+  coordinates: number;
+}
```
Comment (YDrogen) [inline packages/types/src/database.ts:40]: Si c'est pour faire des `Omit`, autant refaire les définitions
Ça fait plus propre, même si le code est dupliqué

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/types/src/database.ts [hunk 2]
```diff
@@ -1,14 +1,139 @@
...
+  id: Generated<number>;
+  order?: number;
+  user_id: number;
+};
+
+export type UserTable = {
+  id: Generated<number>;
+  name?: string;
+  birthdate?: Date;
+  gender?: string;
```
Comment (Mikazian) [inline packages/types/src/database.ts:25]: J'aurai créé un type Gender = male | female | non-binary

**Review by ElBarto444 — DISMISSED** [review]
Bon travail Grégory

File: packages/backend/migrations/src/migrations/20231122143009-database-structure.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+    await trx.schema
+      .createTable('hobby_user')
+      .ifNotExists()
+      .addColumn('id', 'integer', (col) =>
+        col.autoIncrement().primaryKey().unsigned().notNull(),
+      )
+      .addColumn('order', 'integer', (col) => col.unsigned())
+      .addColumn('user_id', 'int4', (col) => col.unsigned())
+      .addColumn('hobby_id', 'int2', (col) => col.unsigned())
```
Comment (Mikazian) [inline packages/backend/migrations/src/migrations/20231122143009-database-structure.ts:188]: integer

**Review by ElBarto444 — APPROVED** [review]

File: packages/types/src/database.ts [hunk 4]
```diff
@@ -1,14 +1,143 @@
...
 }
+export interface LanguageUserTable extends IntermediateTable {
+  language_id: number;
+}
+export interface PictureUserTable extends IntermediateTable {
+  picture_id: number;
+}
+export interface HobbyUserTable extends IntermediateTable {
+  hobby_id: number;
+}
```
Comment (YDrogen) [inline packages/types/src/database.ts:76]: On n'avait pas dit qu'on changeait ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**